docs/proposals/versioning-and-release.md

```diff
@@ -0,0 +1,72 @@
+# Devfile Versioning and Release Process
+This design document outlines the proposed versioning and release process for the Devfile spec. 
+
+This document summarizes parts from he Devfile API technical meeting slides presented by @davidfestal back in October, and I recommend having a read through of it: https://docs.google.com/presentation/d/1ohM1HzPB59a3ajvB7rVWJkKONLBAuT0mb5P20_A6vLs/edit#slide=id.g8fc722bef9_1_8
```

Comment:
typo: "he Devfile API..." should be "the Devfile API..."

Reply:
Updated. My `t` is broken on my laptop so I have to hit extra hard for characters to show up 😅